modules/dag-l1/src/main/scala/org/tessellation/dag/l1/StateChannel.scala

```diff
@@ -218,7 +218,7 @@ class StateChannel[
 
   private val globalSnapshotProcessing: Stream[F, Unit] = Stream
     .awakeEvery(10.seconds)
-    .evalMap(_ => services.globalL0.pullGlobalSnapshots)
+    .evalMap(_ => storages.lastSnapshot.getOrdinal >>= services.globalL0.pullGlobalSnapshots)
```

Comment:
It will now fail for the currency. The `lastSnapshot` stores a snapshot native to the state channel, so for currency it's `CurrencyIncrementalSnapshot` and for DAG it's `GlobalIncrementalSnapshot`. After this change downloading consecutive global snapshots will fail for currency cause the last ordinal is taken from `CurrencyIncrementalSnapshot` and not `GlobalIncrementalSnapshot` like it should. Because of this error I think that changes in `GlobalL0Service` should be reverted.

Reply:
I agree. Fun fact is that it is a leftover because I am not even using `pullGlobalSnapshots` method in rollback but just `pullLatestSnapshot` directly. I am reverting it.